### PR TITLE
Make multiline docstrings start with a newline.

### PR DESCRIPTION
**Pull request by anntzer**
A few are still missing.

Also inherited a couple more docstrings.

Now we just need to figure out how to flake8-enforce that...

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by timhoffm** [comment]
> Now we just need to figure out how to flake8-enforce that...

At least delete D213 ignore from `.flake8`. Not sure if that's enough. It may be that the due to the default docstring convention in pydocstyle/flake8-docstrings D213 is still not checked. In that case one needs to dig through flake8/flake8-docstrings/pydocstyle to find out how to explicitly activate it.

**Comment by anntzer** [comment]
Removing it doesn't enforce the check (I tried), and in any case it's not actually fully fixed yet because there's still a problem with single-line docstrings which don't fit in one line when the quotes are in the same line, but do if the quotes are not there.  This was raised at least once on the pydocstyle tracker (should be easy to find) and last time I check their response was basically "don't write single-line docstrings of just the line that's annoying"...

File: lib/mpl_toolkits/axes_grid1/axes_rgb.py [hunk 1]
```diff
@@ -118,7 +118,8 @@ def __init__(self, *args, pad=0, add_all=True, **kwargs):
         self._config_axes()
 
     def _config_axes(self, line_color='w', marker_edge_color='w'):
-        """Set the line color and ticks for the axes
+        """
+        Set the line color and ticks for the axes
```
Comment (timhoffm) [inline lib/mpl_toolkits/axes_grid1/axes_rgb.py:122]:
```suggestion
        Set the line color and ticks for the axes.
```

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 2]
```diff
@@ -1061,7 +1061,8 @@ def format_coord(self, xd, yd):
         return 'x=%s, y=%s, z=%s' % (xs, ys, zs)
 
     def _on_move(self, event):
-        """Mouse moving
+        """
+        Mouse moving
```
Comment (timhoffm) [inline lib/mpl_toolkits/mplot3d/axes3d.py:1065]:
```suggestion
        Mouse moving.
```

**Comment by anntzer** [comment]
sure, done

**Comment by timhoffm** [comment]
> there's still a problem with single-line docstrings which don't fit in one line when the quotes are in the same line, but do if the quotes are not there.

That's a separate error code: D200. I think D213 still allows
~~~
def foo():
    """
    Single line.
    """
~~~

> This was raised at least once on the pydocstyle tracker (should be easy to find) and last time I check their response was basically "don't write single-line docstrings of just the line that's annoying"...

There was https://github.com/PyCQA/pydocstyle/issues/184 with D205 and D400 errors if you have very long first sentences that need to be wrapped. - IMHO they would be relatively easy to fix, but the issue was closed.

Additionally, I opened https://github.com/PyCQA/pydocstyle/issues/453, because D200 cannot be fulfilled for docstrings with `max. line length - 6 < text length + indent <= max. line length`. This is a bit harder to fix, because pycodestyle would need to know about my line limit to allow for an exception.

For now, this means that we cannot check D200, 205, D400, because I don't think we want to or even reasonably can rewrite the first sentence to be descrpitive and fit in 69 chars (or 65 for methods).

**Comment by anntzer** [comment]
ok, removed D213 from .flake8, we'll see how this goes...

**Comment by timhoffm** [comment]
Just checked PEP-257 again on multi-line docstrings:

> The summary line may be used by automatic indexing tools; it is important that it fits on one line and is separated from the rest of the docstring by a blank line. 

So, if one plays strictly by the rules, it's viable to close PyCQA/pydocstyle#184 and require the authors to instead shorten the docstring. :sad: Though I don't think that's a reasonable way.

**Comment by anntzer** [comment]
I'm not sure how it matters here or even if an exegesis of PEP257 makes sense, but if we're going to go that route note that PEP257 contains the example (verbatim)
```
def foo():
    """
    This is the second line of the docstring.
    """
```
so "triple-quotes + newline + single-line-docstring + newline + triple-quotes" is apparently allowed.

**Pull request merged**